x/protocolpool/abci.go

```diff
@@ -0,0 +1,73 @@
+package protocolpool
+
+import (
+	"fmt"
+
+	"cosmossdk.io/x/protocolpool/keeper"
+	"cosmossdk.io/x/protocolpool/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func EndBlocker(ctx sdk.Context, k keeper.Keeper, authKeeper types.AccountKeeper) error {
+	err := k.BudgetProposal.Walk(ctx, nil, func(key sdk.AccAddress, budget types.BudgetProposal) (stop bool, err error) {
```

Comment:
should we do this lazily where we dont use endblock? its calculated when they claim the funds